### DISCUSSION TITLE
Add or remove elements

### DISCUSSION THREAD

**Discussion by mathisloge** [litehtml/litehtml#259] [Q&A] [answered]
Hey there!

I wanted to dynamicly create or remove elements from the document to not modify strings and parse the whole document again.
I tried 
```cpp
auto new_el = mydocument->create_element("div", {{"class", "clazz"}});
...
mydocument->root()->get_child(kBodyIndex)->appendChild(new_el);
...
doc->render(width, litehtml::render_all);
litehtml::position clip(0, 0, width, height);
doc->draw(0, 0, 0, &clip);
```

But unfortunatly there isn't much happening. Should this work? Do I miss something? 
I've already searched the issues but didn't found anything helpful.

Expection:
The new div element with `class: "clazz"` is drawn.

**Comment by mathisloge** [comment, marked as answer]
Ah... 
So I've now moved https://github.com/litehtml/litehtml/blob/master/src/document.cpp#L80-L132 into its own function and now calling it after the dom updates. That works at least for me very seamlessly.

**Reply by blafed** [reply]
how did you achieve this? I am trying to create elements and append them to body but nothing being shown. even the elements are being represented in the tree, but i guess they are not presented in the render tree 🤷‍♂️

**Reply by mathisloge** [reply]
See https://github.com/litehtml/litehtml/pull/267 but note the comment. Haven't used it since, as I've moved to another solution without litehtml
